base_layer/core/src/transactions/transaction_protocol/sender.rs

```diff
@@ -345,10 +354,15 @@ impl SenderTransactionProtocol {
             let total_amount = info.calculate_total_amount();
             let fee = info.metadata.fee;
             // The fee should be less than the amount. This isn't a protocol requirement, but it's what you want 99.999%
-            // of the time, and our users will thank us if we reject a tx where they put the amount in the fee field by
-            // mistake!
+            // of the time, however, preventing this will also prevent spending dust.
             if fee > total_amount {
-                return Err(TPE::ValidationError("Fee is greater than amount".into()));
+                if !info.dust_collector_enabled {
```

Comment:
` if (!info.dust_collector_enabled && fee > total_amount)..`

and save yourself a nested if

Reply:
Going with this:
``` rust
            if fee > total_amount {
                warn!(
                    target: LOG_TARGET,
                    "Fee ({}) is greater than amount ({}) for Transaction (TxId: {}).", fee, total_amount, info.ids[0]
                );
                if info.prevent_fee_and_amount_swap {
                    return Err(TPE::ValidationError("Fee is greater than amount".into()));
                }
            }
```